### DISCUSSION TITLE
Unable to install Web3uikit

### DISCUSSION THREAD

**Discussion by itGaodazhuang** [web3ui/web3uikit#1074] [Q&A]
Hello, i'm trying to install web3uikit by this order `yarn add  @web3uikit/web3` in my WSL Ubantu system, but it doesn't work, the errors are as follows: 
```
error An unexpected error occurred: "https://github.com/oscario2/wp5-cra-polyfill: ESOCKETTIMEDOUT".
info If you think this is a bug, please open a bug report with the information provided in "/home/zd_lzy/hh-fcc/nextjs-lottery/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

sometime when I try to install it, I would meet this error either: 
```
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote https://github.com/ethereumjs/ethereumjs-abi.git HEAD
Directory: /home/zd_lzy/hh-fcc/nextjs-lottery
Output:
fatal: unable to access 'https://github.com/ethereumjs/ethereumjs-abi.git/': GnuTLS recv error (-110): The TLS connection was non-properly terminated.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```
I had tried to delete the folder node_module and run the order `yarn cache clean`, but these two methods are useless.
what should I do?
ps: I could successfully install the module @web3uikit/web3 and @web3uikit/icons, but not @web3uikit/web3

**Comment by Alpa8820** [comment]
Hi there! 
It looks like you're facing network-related issues while trying to install @web3uikit/web3 via Yarn in your WSL Ubuntu environment. The errors you shared point to network timeouts and TLS connection failures during Git operations. Here's a checklist of things you can try to resolve this:
1. Check Your Internet / VPN / Proxy
Ensure your internet connection is stable.
If you're behind a VPN, corporate proxy, or firewall, try disabling it or switching networks.
If you're using WSL2, make sure Windows isn't limiting WSL's access to the internet.
2. Try Using SSH Instead of HTTPS for Git
If Git over HTTPS is failing, you can try switching to SSH:
git config --global url."git@github.com:".insteadOf "https://github.com/"
This forces Yarn (and Git) to use SSH instead of HTTPS when pulling dependencies.
3. Increase Yarn Network Timeout
Yarn might be failing due to short timeout settings. Increase the timeout:
yarn config set network-timeout 600000
4. Try Installing with NPM Instead
Sometimes Yarn has issues with specific packages. Try:
npm install @web3uikit/web3
5. Update Git and Yarn
Older versions of Git or Yarn might cause TLS-related errors:
sudo apt update && sudo apt upgrade git
npm install -g yarn
Let me know if none of these work and I can help you debug further! 🚀
